#1377: Don't use `Home.rel()` for logging

volodya-lombrozo opened the pull request:
In the most cases we used `new Home().rel(fullPath)` statement without using constructor,  it means that the statement just returned `fullPath` as is, without any changes. 
According with the puzzle, we decided to throw exception if `fullPath` doesn't have common parent directories with `Home.cwd` (see tests and implementation). It means that I had two options:
1) Remove `new Home().rel(fullPath)` statements and use `fullPath` instead - (the logic don't change in that case). And it makes sense, because in logs it's better to see the full path instead of related path (at least it useful for debugging to see the real full path). 
2) Use something like `new Home(workdir).rel(fullPath)`, but we don't have `workdir` and adding it to all classes would be much longer comparing with (1) and the benefits are questionable.
___
by that reason I've decided to go by the (1) way. After implementation - I've noticed that `Home.rel` doesn't using anywhere. By that reason I removed it at all.

Closes: #1377

volodya-lombrozo commented:
@mximp Please, take a look

yegor256 commented:
@volodya-lombrozo the purpose of `Home.rel()` is to make our logs shorter. We don't want to see full path in the log, we only want to see the relative one, related to the current work dir. With your changes we get back to the original problem -- log lines will be long and hard to read.

yegor256 commented:
@volodya-lombrozo we have just "current working dir" from the system. If it's `/user/yegor/eo` and file name is `/user/yegor/eo/test/foo.eo`, then we print `./test/foo.eo` in the log. This is what a user would easily understand. Moreover, the user will be able to copy this string from the log and use in console.

mximp commented:
> @volodya-lombrozo we have just "current working dir" from the system....

@yegor256 it's not always correct: for example in our tests we dynamically shift the base to `/tmp/xxxxx` and "current working dir" remains what it was.
Also when we cache files in `.eo/` folder, which is different base from "current working dir".
So I guess we indeed has different base locations in different cases.

yegor256 commented:
@volodya-lombrozo the _only_ purpose of `.rel()` method is to make logging shorter. When we make file paths shorter in the log, we can't modify them anyhow, except by making them related to _current_ working directory of the running process. Any other modifications will only confuse the user.

My point is that `.rel()` method may/should/must be `static`. Now it's not and this confuses you, I believe.

volodya-lombrozo commented:
@yegor256 

I have a test that set `targetDir` to:
`/var/folders/w0/69c9m3f92t9bbcdw8ptgfjyw0000gn/T/junit11295647479916889676`

My working dir is:

`/Users/lombrozo/Workspace/EOlang/eo`

I have the next code:
```java
new Home().rel(targetDir)
```

What should return that statement?

yegor256 commented:
@volodya-lombrozo in this case, `.rel()` doesn't change the text and returns `/var/folders/w0/69c9m3f92t9bbcdw8ptgfjyw0000gn/T/junit11295647479916889676`

yegor256 commented:
@volodya-lombrozo I think it's a wrong puzzle. I suggest deleting it and making `.rel()` static.

mximp commented:
@volodya-lombrozo @yegor256 The reason for this puzzle was to make `rel()` method natural to `Home` object which is to return relative path to whatever `Home` wraps (not to current working dir). If `rel()` method is not used anywhere in such a context it should be removed from `Home` at all. 
Shortening file paths for logging has no relation to `Home` object in this case. And in this case suggestion to create separate `Rel` class with "relativizing" paths sounds reasonable.

volodya-lombrozo commented:
@mximp, @yegor256 can you review, please? What I have done:
1. Added `Rel` class
2. Added test for `Rel` class - `RelTest` 
4. All `new Home.rel()` invocations were replaced with `Rel.toString`
5. `Home.rel` methods and connected tests were removed

PS: also I had to change `rel` method little bit in order to make all tests pass for Windows OS

mximp commented on eo-maven-plugin/src/test/java/org/eolang/maven/RelTest.java:
```diff
@@ -0,0 +1,59 @@
+class RelTest {
+
+    @ParameterizedTest
+    @CsvSource({
```
@volodya-lombrozo I believe we want to test the case when base and other are from different hierarchies

volodya-lombrozo replied:
@mximp Done

volodya-lombrozo commented:
@yegor256 merge please.

rultor commented:
> @rultor merge

@yegor256 OK, I'll try to merge now. You can check the progress of the merge [here](https://www.rultor.com/t/27818-1302022923)

rultor commented:
> @rultor merge

@yegor256 Done! FYI, the full log is [here](https://www.rultor.com/t/27818-1302022923) (took me 7min)
